projects/fw103_hw/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include <stdint.h>  // for integer types
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+static GpioAddress s_adc_addr = { .port = GPIO_PORT_A, .pin = 6 };
+static GpioAddress s_button_addr = { .port = GPIO_PORT_B, .pin = 2 };
+
+static GpioSettings s_adc_settings = {
+  .direction = GPIO_DIR_IN,
+  .state = GPIO_STATE_LOW,
+  .resistor = GPIO_RES_NONE,
+  .alt_function = GPIO_ALTFN_ANALOG,  // a connection to peripherals
+};
+
+// static GpioSettings s_button_settings = {
+//   .direction = GPIO_DIR_IN,
+//   .resistor = GPIO_RES_PULLDOWN,
+//   .alt_function = GPIO_ALTFN_NONE, // no peripherals
+// };
```

Comment:
Generally remove any commented-out code before pushing, but I think this is OK for the homework.